vault/resource_kv_secret_v2.go

```diff
@@ -220,12 +220,33 @@ func kvSecretV2Read(_ context.Context, d *schema.ResourceData, meta interface{})
 	shouldRead := !d.Get("disable_read").(bool)
 
 	path := d.Id()
+	if path == "" {
+		return nil
+	}
 
 	if err := d.Set(consts.FieldPath, path); err != nil {
 		return diag.FromErr(err)
 	}
 
+	// id should be of the form "mount/data/name"
+	// limit substrings to 3 in case name has '/'
+	// in it or if it's a nested secret
+	parsedPath := strings.SplitN(path, "/", 3)
+
+	mount := parsedPath[0]
+	name := parsedPath[2]
```

Comment:
Should we be checking the length of `parsedPath` before accessing an index? Or can we safely assume that it has a length of 3 here?

Reply:
Good call! The length should be 3 based on how we're constructing it within the `Write` method, but added an assertion check just to be sure 😄